pkg/build/package.go

```diff
@@ -319,6 +319,37 @@ func findInterpreter(bin *elf.File) (string, error) {
 	return "", nil
 }
 
+// dereferenceCrossPackageSymlink attempts to dereference a symlink across multiple package
+// directories.
+func (pc *PackageContext) dereferenceCrossPackageSymlink(path string) (string, error) {
+	libDirs := []string{"lib", "usr/lib", "lib64", "usr/lib64"}
+	targetPackageNames := []string{pc.PackageName, pc.Context.Configuration.Package.Name}
+	realPath, err := os.Readlink(filepath.Join(pc.WorkspaceSubdir(), path))
+	if err != nil {
+		return "", err
+	}
+
+	realPath = filepath.Base(realPath)
+
+	for _, subPkg := range pc.Context.Configuration.Subpackages {
+		targetPackageNames = append(targetPackageNames, subPkg.Name)
+	}
+
+	for _, pkgName := range targetPackageNames {
+		basePath := filepath.Join(pc.Context.WorkspaceDir, "melange-out", pkgName)
+
+		for _, libDir := range libDirs {
+			testPath := filepath.Join(basePath, libDir, realPath)
+
+			if _, err := os.Stat(testPath); err == nil {
```

Comment:
This seems to take the first matching package. Is there any case where multiple subpackages might produce a candidate for the target lib?

I normally think of subpackages as unordered unless one depends on another, and this feels like a relatively unusual/subtle way for subpackages to depend on another in an ordered way.

Maybe we should fail if multiple candidates are found instead of taking the first?

Reply:
or do it in a more predictable way? sort first then take first at least?

Reply:
It does not matter what order the packages are checked in.  All we are looking for is one package which has the target library, so we can learn the actual SONAME.  There would not be a scenario where building multiple variations of the same library results in a different SONAME, but not also a different symlink pattern.